src/mgr/DaemonServer.cc

```diff
@@ -1773,24 +1773,27 @@ void DaemonServer::got_mgr_map()
   Mutex::Locker l(lock);
   set<std::string> have;
   cluster_state.with_mgrmap([&](const MgrMap& mgrmap) {
+      auto md_update = [&] (DaemonKey key) {
+        std::ostringstream oss;
+        auto c = new MetadataUpdate(daemon_state, key);
+        oss << "{\"prefix\": \"mgr metadata\", \"who\": \""
+              << key.second << "\"}";
+        monc->start_mon_command({oss.str()}, {}, &c->outbl, &c->outs, c);
```

Comment:
I think on the osd/mds path we do some checking of whether the metadata is already being fetched in DaemonStateIndex.updating, should do similar here to avoid firing off multiple requests

Reply:
Yes indeed...I wasn't aware of potential races when two standby daemons get the new map. I'm wondering wouldn't it make sense to put the call to ```daemon_state.notify_updating()``` in the ```MetadataUpdate``` constructor? The updating flag is already cleared in ```MetadataUpdate.finish()``` so we might as well set as part of this class.

Reply:
Yeah, anything to move the calls into a common place is :+1: